src/main/java/net/dries007/tfc/common/blocks/crop/CropBlock.java

```diff
@@ -223,7 +232,25 @@ public ClimateRange getClimateRange()
     @Override
     public FarmlandBlockEntity.NutrientType getPrimaryNutrient()
     {
-        return primaryNutrient;
+        return primaryNutrient.get();
+    }
+
+    @Override
+    public float getGrowthFactor(Level level, BlockPos pos, BlockState state, CropBlockEntity crop)
+    {
+        return 1f / baseGrowthTime.get();
+    }
+
+    @Override
+    public float getNutrientConsumptionFactor(Level level, BlockPos pos, BlockState state, CropBlockEntity crop)
+    {
+        return 1f / nutrientConsumptionTime.get();
+    }
+
+    @Override
+    public float getNutrientResupplyFactor(Level level, BlockPos pos, BlockState state, CropBlockEntity crop)
```

Comment:
why in the everloving fuck would you need to know the level, position, block state, and entire block entity to know how much the nutrient could resupply by in one single method call when like I said previously... you could just override that one method.

Reply:
Ex if you wanted nutrients to not resupply as much in the dark

Reply:
so... override that one method. What if you want only phosphorus to only resupply as much in the dark? Then you couldn't do that with this, you'd need to override that one method.